source/secure_tunneling.c

```diff
@@ -128,6 +130,57 @@ static int s_secure_tunneling_close(struct aws_secure_tunnel *secure_tunnel) {
     return AWS_OP_SUCCESS;
 }
 
+static bool s_secure_tunneling_send_data_call(
+    struct aws_websocket *websocket,
+    struct aws_byte_buf *out_buf,
+    void *user_data) {
+    UNUSED(websocket);
+    struct aws_byte_buf *buffer = user_data;
+    AWS_RETURN_ERROR_IF2(aws_byte_buf_write(out_buf, buffer->buffer, buffer->len) == AWS_OP_SUCCESS, false);
+    aws_byte_buf_clean_up(buffer);
+    return true;
+}
+
+static void s_init_websocket_send_frame_options(
+    struct aws_websocket_send_frame_options *frame_options,
+    struct aws_byte_buf *buffer) {
+    frame_options->payload_length = buffer->len;
+    frame_options->user_data = buffer;
+    frame_options->stream_outgoing_payload = s_secure_tunneling_send_data_call;
+    frame_options->opcode = AWS_WEBSOCKET_OPCODE_BINARY;
+    frame_options->fin = true;
+    frame_options->high_priority = false;
+    AWS_ZERO_STRUCT(frame_options->rsv);
```

Comment:
`on_complete` is not set yet. Are you planning to do a followup PR?

Reply:
done